server/async_tcp.go

```diff
@@ -17,247 +16,211 @@ import (
 	"github.com/dicedb/dice/server/utils"
 )
 
-var cronFrequency time.Duration = 1 * time.Second
-var lastCronExecTime time.Time = utils.GetCurrentTime()
-
-const EngineStatusWAITING int32 = 1 << 1
-const EngineStatusBUSY int32 = 1 << 2
-const EngineStatusSHUTTINGDOWN int32 = 1 << 3
-const EngineStatusTRANSACTION int32 = 1 << 4
-
-var eStatus int32 = EngineStatusWAITING
-
-var connectedClients map[int]*core.Client
+type AsyncServer struct {
+	serverFD         int
+	ctx              context.Context
+	cancel           context.CancelFunc
+	wg               *sync.WaitGroup
```

Comment:
Since we need to propagate the wait-group all the way down, it may not be good to store it as part of struct.
If you feel it is complicated we can create a waitgroup that has only one level depth.